include/boost/url/impl/decoded_view.ipp

```diff
@@ -0,0 +1,33 @@
+//
+// Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/url
+//
+
+#ifndef BOOST_URL_IMPL_PCT_DECODED_RANGE_IPP
+#define BOOST_URL_IMPL_PCT_DECODED_RANGE_IPP
+
+#include <boost/url/decoded_view.hpp>
+#include <memory>
+#include <new>
+
+namespace boost {
+namespace urls {
+
+std::ostream&
+operator<<(std::ostream& os,
+    decoded_view const& str) {
+    auto it = str.begin();
+    auto end = str.end();
+    for (; it != end; ++it)
+        os.put(*it);
+    return os;
+}
+
+} // urls
```

Comment:
LOL WUT??? Where's the rest of the file? A goal of our new libraries is _fast compilation time_. That's why we prefer designs that don't require templates, and we take great care to put as many function definitions as possible out-of-line. It looks like we have just slapped "constexpr" on everything possible without really evaluating the costs and benefits. I don't see much of a use-case for comparing lazily percent-decoded ranges at compile time if I'm being honest. So I don't think it is a good decision to mark those things constexpr and then force everyone in the universe to recompile that function every time the header is included.

Contrary to the prevailing Reddit wisdom, we definitely should not "constexpr all the things." Instead we should be thoughtful about where and how we mark things constexpr. There should be an articulable and compelling benefit to its use.